app/scripts/background.js

```diff
@@ -707,6 +707,9 @@ export function setupController(
     } else {
       // this is triggered when a new tab is opened, or origin(url) is changed
       if (remotePort.sender && remotePort.sender.tab && remotePort.sender.url) {
+        console.log(`delaying connection to ${remotePort.sender.url} for 5s`);
+        await new Promise((resolve) => setTimeout(resolve, 5000));
+
```

Comment:
this is here to make testing easier. It delays the handling of the dapp responses over the port for 5s.

Reply:
TODO: Remove this before approval, of course

Reply:
This has been removed, but must be added back locally if you are testing manually